tests/bug00990.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+Test for bug #990: DBGP: Add notification for notices, warnings and errors
+--SKIPIF--
+<?php if (getenv("SKIP_DBGP_TESTS")) { exit("skip Excluding DBGp tests"); } ?>
+--INI--
+xdebug.remote_enable=1
+--FILE--
+<?php
+require 'dbgp/dbgpclient.php';
+$data = file_get_contents(dirname(__FILE__) . '/bug00990.inc');
+
+$commands = array(
+	'step_into',
+	'step_into',
+	'feature_set -n notify_ok -v 1',
+	'step_into',
+	'detach'
+);
+
+dbgpRun( $data, $commands );
+?>
+--EXPECTF--
+<?xml version="1.0" encoding="iso-8859-1"?>
+<init xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" fileuri="file:///tmp/xdebug-dbgp-test.php" language="PHP" xdebug:language_version="" protocol_version="1.0" appid="" idekey=""><engine version=""><![CDATA[Xdebug]]></engine><author><![CDATA[Derick Rethans]]></author><url><![CDATA[http://xdebug.org]]></url><copyright><![CDATA[Copyright (c) 2002-2099 by Derick Rethans]]></copyright></init>
+
+-> step_into -i 1
+<?xml version="1.0" encoding="iso-8859-1"?>
+<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="step_into" transaction_id="1" status="break" reason="ok"><xdebug:message filename="file:///tmp/xdebug-dbgp-test.php" lineno="2"></xdebug:message></response>
+
+-> step_into -i 2
+<?xml version="1.0" encoding="iso-8859-1"?>
+<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="step_into" transaction_id="2" status="break" reason="ok"><xdebug:message filename="file:///tmp/xdebug-dbgp-test.php" lineno="3"></xdebug:message></response>
+
+-> feature_set -i 3 -n notify_ok -v 1
+<?xml version="1.0" encoding="iso-8859-1"?>
+<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="feature_set" transaction_id="3" feature="notify_ok" success="1"></response>
+
+-> step_into -i 4
+<?xml version="1.0" encoding="iso-8859-1"?>
+<notify xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" name="debug"><xdebug:message filename="file:///tmp/xdebug-dbgp-test.php" lineno="3" type="Notice"><![CDATA[Undefined variable: bar]]></xdebug:message><![CDATA[Notice: Undefined variable: bar in /tmp/xdebug-dbgp-test.php on line 3]]></notify>
```

Comment:
"debug" should be "error" here.

Xdebug should also add a stack trace for notifications as normally, and rewrite the stack trace in case of an exception as it does this for normal exceptions too: https://github.com/xdebug/xdebug/blob/46dccd6a50512d9f41cc9932afe04afd18e3f86f/xdebug_stack.c#L693